src/main/java/oop/course/routes/LoginRoute.java

```diff
@@ -1,20 +1,48 @@
 package oop.course.routes;
 
+import oop.course.entity.Customer;
+
 import oop.course.interfaces.Process;
 import oop.course.interfaces.*;
+import oop.course.responses.UnauthorizedResponse;
+import oop.course.tools.implementations.JsonForm;
+import oop.course.tools.interfaces.Form;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.Connection;
 
 public class LoginRoute implements Route {
     private final Process next;
+    private final Connection connection;
+    private final Logger log = LoggerFactory.getLogger(LoginRoute.class);
 
-    public LoginRoute(Process next) {
+    public LoginRoute(Connection connection, Process next) {
+        this.connection = connection;
         this.next = next;
     }
 
     @Override
-    public Response act(Request request) {
+    public Response act(Request request)  {
+        Form form = new JsonForm(request.body());
+        Customer customer = new Customer(connection, form);
+        if (!customer.exists()) {
+            log.error("Customer not found:\nEmail: " +
```

Comment:
not extract from form. Let customer tell that his email or whatever does not exist. Just check the flag